CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8.12)
```

Comment:
https://github.com/google/benchmark/pull/717#issuecomment-438266258
> As i have posted in #613, unfortunately, now there is a small roadblock here:
> https://github.com/llvm-mirror/llvm/blob/c1873bf60aa20b3873d859c99ecaf5967c20ba3f/CMakeLists.txt#L3
> 
> ```cmake
> cmake_minimum_required(VERSION 3.4.3)
> ```

Can we go only up to 3.4.3?

Reply:
i'm in two minds. on the one hand, any bump is better than no bump. on the other hand, why would we use llvm as a blocker? llvm may require 3.4.3, but they also support >3.4.3. on my local machine, the current version is 3.12.1 so support is broad for higher versions.

Reply:
Version requirement is two-fold. If project A requires version X,
and depends on project B that requires version Y, and Y > X,
then project A effectively requires version Y.
In other words, this will bump the version required by LLVM.

That being said, cmake-v3.5.x is 3 years old already (i know magical 3 years :D), so LG.
https://github.com/Kitware/CMake/tags?after=v3.5.2